groups/nts/ntsa/ntsa_localname.h

```diff
@@ -42,16 +42,21 @@ namespace ntsa {
 class LocalName
 {
   public:
-    enum {
-        /// The maximum path length, not including the null terminator.
-        k_MAX_PATH_LENGTH = 92
-    };
+    /// The maximum path length, in case of abstract namespace it does not
+    /// include leading null, in case of real namespace it does not include the
+    /// null terminator
+
+    static const bsl::size_t k_MAX_PATH_LENGTH;
```

Comment:
I like all the changes you want to do in this PR except for those around the representation of the storage of the UDS name and its capacity. As proposed, `k_CAPACITY` now explodes the size of `ntsa::Endpoint` on AIX and is not representable in the `bsl::uint8_t d_size`. I'm also unconvinced about the correctness 
 or usefulness of `k_MAX_PATH_LENGTH` and `k_CAPACITY`, let's discuss this more so I can understand what you want to do.